### ISSUE TITLE
Tweaking horovod: Sharing weights instead of gradients

### ISSUE THREAD

**Issue by SrivastavaKshitij** [horovod/horovod#286]
Hi @alsrgv 

I am trying to use horovod in a different way and for that I need to all-reduce weights of layers instead of gradients. Is there a way I can do that. I am using horovod + pytorch on Summit supercomputer. 

Thanks
Kshitij Srivastava
ORNL

**Comment by alsrgv** [comment]
@SrivastavaKshitij, great to see one of the first Horovod + PyTorch usages! You should be able to do something along the lines of:
```python
...
output = model(data)
loss = F.nll_loss(output, target)
loss.backward()
optimizer.step()

handles = []
# allreduce in-place asynchronously
for name, p in model.named_parameters():
  handle = hvd.allreduce_async_(p.data, average=True, name=name)
  handles.append(handle)
# synchronize
for handle in handles:
  hvd.synchronize(handle)
```

**Comment by SrivastavaKshitij** [comment]
Thanks @alsrgv 👍
